include/aspect/newton.h

```diff
@@ -113,88 +113,129 @@ namespace aspect
-                                                 internal::Assembly::CopyData::StokesSystem<dim> &data) const;
+        execute (internal::Assembly::Scratch::ScratchBase<dim>  &scratch,
+                 internal::Assembly::CopyData::CopyDataBase<dim> &data) const;
+    };
 
-        /**
-         * This function assembles the right-hand-side term of the Newton Stokes system
-         * that is caused by the compressibility in the mass conservation equation.
-         * This function approximates this term as
-         * $- \nabla \mathbf{u} = \frac{1}{\rho} * \frac{\partial rho}{\partial z} \frac{\mathbf{g}}{||\mathbf{g}||} \cdot \mathbf{u}$
-         */
+    /**
+     * This class assembles the term that arises in the viscosity term of the Newton Stokes matrix for
+     * compressible models, because the divergence of the velocity is not longer zero.
+     */
+    template <int dim>
+    class NewtonStokesCompressibleStrainRateViscosityTerm : public Assemblers::Interface<dim>,
+      public SimulatorAccess<dim>
+    {
+      public:
+        virtual ~NewtonStokesCompressibleStrainRateViscosityTerm () {};
+
+        virtual
         void
-        reference_density_compressibility_term (const double                                     pressure_scaling,
-                                                const bool                                       rebuild_stokes_matrix,
-                                                internal::Assembly::Scratch::StokesSystem<dim>  &scratch,
-                                                internal::Assembly::CopyData::StokesSystem<dim> &data,
-                                                const Parameters<dim> &parameters) const;
+        execute(internal::Assembly::Scratch::ScratchBase<dim>  &scratch,
+                internal::Assembly::CopyData::CopyDataBase<dim> &data) const;
+    };
 
-        /**
-         * This function assembles the compressibility term of the Newton Stokes system
-         * that is caused by the compressibility in the mass conservation equation.
-         * It includes this term implicitly in the matrix,
-         * which is therefore not longer symmetric.
-         * This function approximates this term as
-         * $ - \nabla \mathbf{u} - \frac{1}{\rho} * \frac{\partial rho}{\partial z} \frac{\mathbf{g}}{||\mathbf{g}||} \cdot \mathbf{u} = 0$
-         */
+    /**
+     * This class assembles the right-hand-side term of the Newton Stokes system
+     * that is caused by the compressibility in the mass conservation equation.
+     * This function approximates this term as
+     * $- \nabla \mathbf{u} = \frac{1}{\rho} * \frac{\partial rho}{\partial z} \frac{\mathbf{g}}{||\mathbf{g}||} \cdot \mathbf{u}$
+     */
+    template <int dim>
+    class NewtonStokesReferenceDensityCompressibilityTerm : public Assemblers::Interface<dim>,
+      public SimulatorAccess<dim>
+    {
+      public:
+        virtual ~NewtonStokesReferenceDensityCompressibilityTerm () {};
```

Comment:
remove semicolon. Do we normally have all those in the .cc files? I don't mind having them here...